### PR TITLE
Generic OpenAI fn chain

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
i dont think we want a chain that also calls the function. i think we just want easy ways to create LLMChains that utilize openai fucnctions

File: langchain/chains/openai_functions/base.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+PYTHON_TO_JSON_TYPES = {
+    "str": "string",
+    "int": "number",
+    "float": "number",
+    "bool": "boolean",
+}
+
+
+def convert_python_function_to_openai_function(function: Callable) -> Dict[str, Any]:
```
Comment (jacoblee93) [inline langchain/chains/openai_functions/base.py:30]: Pretty cool that you can do that

File: langchain/chains/openai_functions/base.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            "name": schema["title"],
+            "description": schema["description"],
+            "parameters": schema,
+        }
+    elif callable(function):
+        return convert_python_function_to_openai_function(function)
+
+    else:
+        raise ValueError(
+            f"Unsupposerted function type {type(function)}. Functions must be passed in as Dict, pydantic.BaseModel, or Callable."
```
Comment (jacoblee93) [inline langchain/chains/openai_functions/base.py:103]: Typo: Unsupposerted

File: docs/extras/modules/chains/popular/openai_functions.ipynb
Comment (baskaryan) [inline docs/extras/modules/chains/popular/openai_functions.ipynb:1]: can drop or move the running function example if we prefer

**Pull request merged**